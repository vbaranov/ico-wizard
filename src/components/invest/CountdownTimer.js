import React from 'react'
import ReactCountdownClock from 'react-countdown-clock'

const CountdownTimer = ({
  displaySeconds,
  nextTick,
  tiersLength,
  days,
  hours,
  minutes,
  seconds,
  msToNextTick,
  onComplete,
  isFinalized
}) => {
  const countdownClock = (
    <div>
      {displaySeconds
        ? null
        : <div className="timer-i">
          <div className="timer-count">{days}</div>
          <div className="timer-interval">Days</div>
        </div>
      }
      {displaySeconds
        ? null
        : <div className="timer-i">
          <div className="timer-count">{hours}</div>
          <div className="timer-interval">Hours</div>
        </div>
      }
      <div className="timer-i">
        <div className="timer-count">{minutes}</div>
        <div className="timer-interval">Mins</div>
      </div>
      {!displaySeconds
        ? null
        : <div className="timer-i">
          <div className="timer-count">{seconds}</div>
          <div className="timer-interval">Secs</div>
        </div>
      }
    </div>
  )

  let message = null

  if (isFinalized) {
    message = 'crowdsale has been finalized'
    msToNextTick = 0
  } else {
    if (nextTick.type) {
      if (nextTick.type === 'start') {
        message = `to start of tier ${nextTick.order} of ${tiersLength}`
      } else {
        message = `to end of tier ${nextTick.order} of ${tiersLength}`
      }
    } else {
      message = 'crowdsale has ended'
    }
  }

  return (
    <div className="timer-container">
      <div style={{ marginLeft: '-20px', marginTop: '-20px' }}>
        <ReactCountdownClock
          seconds={msToNextTick / 1000}
          color="#733EAB"
          alpha={0.9}
          size={270}
          showMilliseconds={false}
          onComplete={onComplete}
        />
      </div>
      <div className="timer">
        <div className="timer-inner">
          {isFinalized ? null : countdownClock}
          <div className="timer-i">
            <div className="timer-interval">
              <strong>
                { message }
              </strong>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}


export default CountdownTimer
