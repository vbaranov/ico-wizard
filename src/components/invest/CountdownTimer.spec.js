import React from 'react'
import CountdownTimer from './CountdownTimer'
import Adapter from 'enzyme-adapter-react-15'
import { configure, shallow } from 'enzyme'

configure({ adapter: new Adapter() })

describe('CountdownTimer', () => {
  it('Should render the component', () => {
    const wrapper = shallow(
      <CountdownTimer
        displaySeconds={false}
        nextTick={{ type: 'start', order: 1 }}
        tiersLength={2}
        days={1}
        hours={4}
        minutes={45}
        seconds={30}
        msToNextTick={30445}
        onComplete={jest.fn()}
        isFinalized={false}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('Should render the component showing only the seconds', () => {
    const wrapper = shallow(
      <CountdownTimer
        displaySeconds={true}
        nextTick={{ type: 'start', order: 1 }}
        tiersLength={2}
        days={1}
        hours={4}
        minutes={45}
        seconds={30}
        msToNextTick={30445}
        onComplete={jest.fn()}
        isFinalized={false}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('Should render the component showing the time until the end of the tier', () => {
    const wrapper = shallow(
      <CountdownTimer
        displaySeconds={false}
        nextTick={{ type: 'end', order: 1 }}
        tiersLength={2}
        days={1}
        hours={4}
        minutes={45}
        seconds={30}
        msToNextTick={30445}
        onComplete={jest.fn()}
        isFinalized={false}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('Should render the component when the crowdsale ended', () => {
    const wrapper = shallow(
      <CountdownTimer
        displaySeconds={false}
        nextTick={{}}
        tiersLength={2}
        days={1}
        hours={4}
        minutes={45}
        seconds={30}
        msToNextTick={30445}
        onComplete={jest.fn()}
        isFinalized={false}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('Should render the component when the crowdsale was finalized', () => {
    const wrapper = shallow(
      <CountdownTimer
        displaySeconds={false}
        nextTick={{ type: 'start', order: 1 }}
        tiersLength={2}
        days={1}
        hours={4}
        minutes={45}
        seconds={30}
        msToNextTick={30445}
        onComplete={jest.fn()}
        isFinalized={true}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it(`Should stop countdown if crowdsale was finalized`, () => {
    const wrapper = shallow(
      <CountdownTimer
        displaySeconds={false}
        nextTick={{ type: 'start', order: 1 }}
        tiersLength={2}
        days={1}
        hours={4}
        minutes={45}
        seconds={30}
        msToNextTick={30445000}
        onComplete={jest.fn()}
        isFinalized={true}
      />
    )

    expect(wrapper.find('ReactCountdownClock').props().seconds).toBe(0)
  })
})
