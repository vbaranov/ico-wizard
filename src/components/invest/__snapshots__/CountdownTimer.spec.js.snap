// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountdownTimer Should render the component 1`] = `
<div
  className="timer-container"
>
  <div
    style={
      Object {
        "marginLeft": "-20px",
        "marginTop": "-20px",
      }
    }
  >
    <ReactCountdownClock
      alpha={0.9}
      color="#733EAB"
      font="Arial"
      fontSize="auto"
      onComplete={[Function]}
      paused={false}
      seconds={30.445}
      showMilliseconds={false}
      size={270}
      timeFormat="hms"
    />
  </div>
  <div
    className="timer"
  >
    <div
      className="timer-inner"
    >
      <div>
        <div
          className="timer-i"
        >
          <div
            className="timer-count"
          >
            1
          </div>
          <div
            className="timer-interval"
          >
            Days
          </div>
        </div>
        <div
          className="timer-i"
        >
          <div
            className="timer-count"
          >
            4
          </div>
          <div
            className="timer-interval"
          >
            Hours
          </div>
        </div>
        <div
          className="timer-i"
        >
          <div
            className="timer-count"
          >
            45
          </div>
          <div
            className="timer-interval"
          >
            Mins
          </div>
        </div>
      </div>
      <div
        className="timer-i"
      >
        <div
          className="timer-interval"
        >
          <strong>
            to start of tier 1 of 2
          </strong>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CountdownTimer Should render the component showing only the seconds 1`] = `
<div
  className="timer-container"
>
  <div
    style={
      Object {
        "marginLeft": "-20px",
        "marginTop": "-20px",
      }
    }
  >
    <ReactCountdownClock
      alpha={0.9}
      color="#733EAB"
      font="Arial"
      fontSize="auto"
      onComplete={[Function]}
      paused={false}
      seconds={30.445}
      showMilliseconds={false}
      size={270}
      timeFormat="hms"
    />
  </div>
  <div
    className="timer"
  >
    <div
      className="timer-inner"
    >
      <div>
        <div
          className="timer-i"
        >
          <div
            className="timer-count"
          >
            45
          </div>
          <div
            className="timer-interval"
          >
            Mins
          </div>
        </div>
        <div
          className="timer-i"
        >
          <div
            className="timer-count"
          >
            30
          </div>
          <div
            className="timer-interval"
          >
            Secs
          </div>
        </div>
      </div>
      <div
        className="timer-i"
      >
        <div
          className="timer-interval"
        >
          <strong>
            to start of tier 1 of 2
          </strong>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CountdownTimer Should render the component showing the time until the end of the tier 1`] = `
<div
  className="timer-container"
>
  <div
    style={
      Object {
        "marginLeft": "-20px",
        "marginTop": "-20px",
      }
    }
  >
    <ReactCountdownClock
      alpha={0.9}
      color="#733EAB"
      font="Arial"
      fontSize="auto"
      onComplete={[Function]}
      paused={false}
      seconds={30.445}
      showMilliseconds={false}
      size={270}
      timeFormat="hms"
    />
  </div>
  <div
    className="timer"
  >
    <div
      className="timer-inner"
    >
      <div>
        <div
          className="timer-i"
        >
          <div
            className="timer-count"
          >
            1
          </div>
          <div
            className="timer-interval"
          >
            Days
          </div>
        </div>
        <div
          className="timer-i"
        >
          <div
            className="timer-count"
          >
            4
          </div>
          <div
            className="timer-interval"
          >
            Hours
          </div>
        </div>
        <div
          className="timer-i"
        >
          <div
            className="timer-count"
          >
            45
          </div>
          <div
            className="timer-interval"
          >
            Mins
          </div>
        </div>
      </div>
      <div
        className="timer-i"
      >
        <div
          className="timer-interval"
        >
          <strong>
            to end of tier 1 of 2
          </strong>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CountdownTimer Should render the component when the crowdsale ended 1`] = `
<div
  className="timer-container"
>
  <div
    style={
      Object {
        "marginLeft": "-20px",
        "marginTop": "-20px",
      }
    }
  >
    <ReactCountdownClock
      alpha={0.9}
      color="#733EAB"
      font="Arial"
      fontSize="auto"
      onComplete={[Function]}
      paused={false}
      seconds={30.445}
      showMilliseconds={false}
      size={270}
      timeFormat="hms"
    />
  </div>
  <div
    className="timer"
  >
    <div
      className="timer-inner"
    >
      <div>
        <div
          className="timer-i"
        >
          <div
            className="timer-count"
          >
            1
          </div>
          <div
            className="timer-interval"
          >
            Days
          </div>
        </div>
        <div
          className="timer-i"
        >
          <div
            className="timer-count"
          >
            4
          </div>
          <div
            className="timer-interval"
          >
            Hours
          </div>
        </div>
        <div
          className="timer-i"
        >
          <div
            className="timer-count"
          >
            45
          </div>
          <div
            className="timer-interval"
          >
            Mins
          </div>
        </div>
      </div>
      <div
        className="timer-i"
      >
        <div
          className="timer-interval"
        >
          <strong>
            crowdsale has ended
          </strong>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CountdownTimer Should render the component when the crowdsale was finalized 1`] = `
<div
  className="timer-container"
>
  <div
    style={
      Object {
        "marginLeft": "-20px",
        "marginTop": "-20px",
      }
    }
  >
    <ReactCountdownClock
      alpha={0.9}
      color="#733EAB"
      font="Arial"
      fontSize="auto"
      onComplete={[Function]}
      paused={false}
      seconds={0}
      showMilliseconds={false}
      size={270}
      timeFormat="hms"
    />
  </div>
  <div
    className="timer"
  >
    <div
      className="timer-inner"
    >
      <div
        className="timer-i"
      >
        <div
          className="timer-interval"
        >
          <strong>
            crowdsale has been finalized
          </strong>
        </div>
      </div>
    </div>
  </div>
</div>
`;
